### ISSUE TITLE
No external SD card over sftp

### ISSUE THREAD

**Issue by fabianrost84** [wolpi/prim-ftpd#51]
I'm have a Motorola Mote E running Android 5.1. I start ftp and sftp server with prim-ftpd. The Moto E emulates an SD card in the internal memory and mounts that at `/storage/sdcard0` and the external SD card is mounted at `/storage/sdcard1`. If I connect (using FileZilla on Kubuntu 15.10) via ftp I can browse both folders. But if I connect via sftp I can only browse the internal memory (`/storage/sdcard0`) but I cannot access the external SD card (`/storage/sdcard1`). The FileZilla log says:

```
Status:    Connecting to 10.42.0.55:1234...
Response:    fzSftp started, protocol_version=2
Command:    open "fabian@10.42.0.55" 1234
Command:    Pass: **************
Status:    Connected to 10.42.0.55
Status:    Retrieving directory listing of "/storage"...
Command:    cd "/storage"
Response:    New directory is: "/storage"
Command:    ls
Status:    Listing directory /storage
Status:    Directory listing of "/storage" successful
Status:    Retrieving directory listing of "/storage/sdcard1"...
Command:    cd "/storage/sdcard1"
Response:    New directory is: "/storage/sdcard1"
Command:    ls
Status:    Listing directory /storage/sdcard1
Error:    Reading directory .: failure
Status:    Directory listing of "/storage/sdcard1" successful
```

**Comment by fabianrost84** [comment]
Today, I set up a sftp server with [SSH Server](https://play.google.com/store/apps/details?id=com.icecoldapps.sshserver). I have the same issue with that sftp server. Maybe this is a hint that the issue is not related to prim-ftpd.

**Comment by wolpi** [comment]
I'm pretty sure that belongs to #31. It will probably take some month until I had enough time to implement that.

**Comment by dandv** [comment]
Same here - Samsung GS7, no root. Saw the same problem on Marshmallow and now on N: when connecting via <b>S</b>FTP, the SD card doesn't appear in the directory listing. Filezilla echoes this error:

> Error:	Directory /storage/emulated: permission denied
> Error:	Failed to retrieve directory 

If I connect over insecure FTP, I do see the SD card directory, but it's not writable.

**Comment by wolpi** [comment]
Interesting that both servers react differently to this. Root cause is the same: apps don't have permission to access sdcard with filesystem api since android 5. See #31.

> I was able to upload over TLS files with the WiFi FTP Server app (non-OSS).

That one surly has implemented SAF support as described in #31.

**Comment by wolpi** [comment]
as #31 is closed this should be done as well
